components/cli/pkg/registry/credentials/credentials.go

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package credentials
+
+import (
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/dgrijalva/jwt-go"
+	"golang.org/x/crypto/ssh/terminal"
+
+	"github.com/cellery-io/sdk/components/cli/pkg/config"
+	"github.com/cellery-io/sdk/components/cli/pkg/util"
+)
+
+// FromBrowser requests the credentials from the user
+func FromBrowser(username string) (string, string, error) {
+	auth := config.LoadConfig().AuthConf
+	timeout := make(chan bool)
+	ch := make(chan string)
+	var code string
+	httpPortString := ":" + strconv.Itoa(auth.CallBackDefaultPort)
+	var codeReceiverPort = auth.CallBackDefaultPort
+	// This is to start the CLI auth in a different port is the default port is already occupied
+	for {
+		_, err := net.Dial("tcp", httpPortString)
+		if err != nil {
+			break
+		}
+		codeReceiverPort++
+		httpPortString = ":" + strconv.Itoa(codeReceiverPort)
+	}
+	redirectUrl := url.QueryEscape("http://" + auth.CallBackHost + ":" + strconv.Itoa(codeReceiverPort) + "/auth")
+	var googleAuthUrl = "https://" + auth.IdpHost + ":" + strconv.Itoa(auth.IdpPort) +
```

Comment:
You can use fmt.Sprintf or url package to build the url

Reply:
Fixed by #1088